src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/LeoRoutes.scala

```diff
@@ -50,37 +50,45 @@ abstract class LeoRoutes(val leonardoService: LeonardoService, val proxyService:
       path("cluster" / Segment / Segment) { (googleProject, clusterName) =>
         put {
           entity(as[ClusterRequest]) { cluster =>
-            complete {
-              leonardoService.createCluster(userInfo, GoogleProject(googleProject), ClusterName(clusterName), cluster).map { cluster =>
-                StatusCodes.OK -> cluster
+            setTokenCookie(userInfo) {
```

Comment:
Since it's shared by all paths, could we put `setTokenCookie` at a higher level, maybe after `requireUserInfo`?

Reply:
Totally. Moving `setTokenCookie` to the top would mean we set the cookie for the `isWhiteListed` API as well. That's fine, right?